From b2d6cbfeb42507650398d02c412ed4f4f346a5cc Mon Sep 17 00:00:00 2001
From: Zhang Jiejing <jiejing.zhang@freescale.com>
Date: Thu, 23 Dec 2010 21:10:53 +0800
Subject: [PATCH 103/498] ENGR00137340 Add headphone detection for mx5x.

Since this route code always noting to do,
it will cause the alsa route not change even the headphone
plugin event already received by HeadphoneObserver.java
so we just open alsa device every time.

Signed-off-by: Zhang Jiejing <jiejing.zhang@freescale.com>
---
 modules/alsa/alsa_imx5x.cpp | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/modules/alsa/alsa_imx5x.cpp b/modules/alsa/alsa_imx5x.cpp
index 1d75161..e2d2598 100755
--- a/modules/alsa/alsa_imx5x.cpp
+++ b/modules/alsa/alsa_imx5x.cpp
@@ -510,7 +510,8 @@ static status_t s_route(alsa_handle_t *handle, uint32_t devices, int mode)
     status_t status = NO_ERROR;
 
     LOGD("route called for devices %08x in mode %d...", devices, mode);
-
+    // below Always noting to do, so we open device every time.
+#if 0
     if (handle->handle && handle->curDev == devices && handle->curMode == mode)
         ; // Nothing to do
     else if (handle->handle && (handle->devices & devices))
@@ -519,7 +520,10 @@ static status_t s_route(alsa_handle_t *handle, uint32_t devices, int mode)
         LOGE("Why are we routing to a device that isn't supported by this object?!?!?!?!");
         status = s_open(handle, devices, mode);
     }
-
+#else
+    setAlsaControls(handle, devices, mode);
+    status = s_open(handle, devices, mode);
+#endif
     return status;
 }
 
-- 
1.8.0

